app/src/main/java/org/kiwix/kiwixmobile/history/HistoryAdapter.java

```diff
@@ -60,7 +67,29 @@ public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int positi
       item.itemView.setOnLongClickListener(v ->
           itemClickListener.onItemLongClick(item.favicon, history));
     } else {
-      ((Category) holder).date.setText(historyList.get(position + 1).getDate());
+
+      String date=historyList.get(position + 1).getDate();
```

Comment:
There should space before and after `=`. You should reformat your code using Android Studio before every commit so that you don't repeat these mistakes.

Reply:
Also, you can remove the blank line before this line.

Reply:
@abdulwd Thanks brother. I did reformatted it but I forgot double-checking it. I've reformatted it again now.

Reply:
> @iadeelzafar the porposal i made was the whole change. would this work?

@soloturn Yes. It will use the latest API that is LocalDate only in devices with API Level 26 or greater and for devices below 26 it will use the Calendar class.